src/main/java/seedu/club/logic/commands/ClearCommand.java

```diff
@@ -11,22 +11,43 @@
 /**
  * Clears the club book.
  */
-public class ClearCommand extends UndoableCommand {
+public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
     public static final String MESSAGE_SUCCESS = "Club Connect has been cleared.";
+    public static final String MESSAGE_FAILURE = "Action to Clear data in Club Connect has been cancelled";
+    public static final String MESSAGE_CONFRIMATION = "Confirm clearing all data in Club Connect?"
+            + " Type 'clear Y' to confirm and 'clear N' to cancel."
+            + " \nWARNING: THIS IS NOT A UNDOABLE COMMAND";
     public static final ArrayList<String> COMMAND_ALIASES = new ArrayList<>(
             Arrays.asList(COMMAND_WORD, "c", "erase")
     );
+    private String args;
 
+    public ClearCommand() {
+    }
+
+    public ClearCommand(String args) {
+        this.args = args;
+    }
 
     @Override
-    public CommandResult executeUndoableCommand() throws CommandException {
+    public CommandResult execute() throws CommandException {
         requireNonNull(model);
         requireToSignUp();
         requireToLogIn();
-        model.resetData(new ClubBook());
-        model.clearClubBook();
-        return new CommandResult(MESSAGE_SUCCESS);
+        if (!model.getClearConfirmation()) {
+            model.setClearConfirmation(true);
+            return new CommandResult(MESSAGE_CONFRIMATION);
+        } else {
+            if (args.equals(" Y")) {
+                model.resetData(new ClubBook());
+                model.clearClubBook();
+                return new CommandResult(MESSAGE_SUCCESS);
+            } else {
+                model.setClearConfirmation(false);
```

Comment:
Perhaps this implementation can be improved in future versions. Right now if I type clear then type some other command      e.g list or add, setClearConfirmation will still remain true. so if accidenttally type clear Y then clubbook will be cleared

Reply:
i've tried a few ways but i think this is the best implementation. I'll state it in the userdoc